contracts/semi-bridge/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use ink_lang as ink;
+
+#[ink::contract(env = pink_extension::PinkEnvironment)]
+mod semi_bridge {
+    use alloc::{string::String, string::ToString, vec::Vec};
+    use index::prelude::*;
+    use index::utils::ToArray;
+    use ink_storage::traits::{PackedLayout, SpreadLayout};
+    use pink_web3::ethabi::{Bytes, Uint};
+    use pink_web3::keys::pink::KeyPair;
+    use pink_web3::signing::Key;
+    use primitive_types::{H160, H256, U256};
+    use scale::{Decode, Encode};
+
+    #[ink(storage)]
+    pub struct SemiBridge {
+        owner: AccountId,
+        key: [u8; 32],
+        config: Option<Config>,
+    }
+
+    #[derive(Encode, Decode, Debug, PackedLayout, SpreadLayout)]
+    #[cfg_attr(
+        feature = "std",
+        derive(scale_info::TypeInfo, ink_storage::traits::StorageLayout)
+    )]
+    struct Config {
+        rpc: String,
```

Comment:
it would be more clear if you add some code here to show what you think the `Config` struct should be like.

actually the code is copied from Hang's rollup project, it's more like an ad hoc solution for a specific application, if we decide to add a strict definition of what an index-contract should look like, let's do that in a particular PR. And, it's best for us to have an RFC describing all the restrictions in writing release-version index contracts

Reply:
It's no matter how we define Config here, the current definition is fine. Just remind us the RPC essentially comes from a chain contract, we can get RPC by <xxx chain contract>.endpoint() because every chain will implement [ChainInspector](https://github.com/Phala-Network/index-contract/blob/27bc4a540fedc4e9a423f167a3e942ebb75c1ccd/traits/registry.rs#L124)

Sure an RFC is better, for now, I guess I didn't get time to do this